### DISCUSSION TITLE
Why analyze so many files(140000), compile_commands.json always only has 10535 compiled objects

### DISCUSSION THREAD

**Discussion by leon0625** [microsoft/vscode-cpptools#10653] [Q&A]
My configuration is as follows
workspace configure: 
```json
{
	"folders": [
		{
			"path": "../ct5gcpe"
		}
	],
	"settings": {}
}
```
cpptool configure:
```json
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${default}"
            ],
            "defines": [],
            "compilerPath": "/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/usr/bin/arm-buildroot-linux-gnueabi-gcc",
            "cStandard": "c99",
            "cppStandard": "gnu++98",
            "compileCommands": "${workspaceFolder}/compile_commands.json",
            "intelliSenseMode": "linux-gcc-x64"
        }
    ],
    "version": 4
}
```
![image](https://user-images.githubusercontent.com/20039046/224247078-7f939fa5-87c7-4d5d-ac25-da57d2898778.png)

```
[ct5gcpe]$ fd . |wc -l
215314
[ct5gcpe]$ grep -nr "{" compile_commands.json |wc -l
10535
```

**The ct5gcpe directory has 200,000 files, shouldn't cpptools only analyze the files in my compile_commands.json ？**

**Comment by sean-mcmanus** [comment]
The with C_Cpp.loggingLevel to set "Debug" the logging shows which folders it looks for files and global symbols in. You could add folders to C_Cpp.files.exclude to exclude more folders. It may process other files from your system include folders, although it won't parse them unless it's matches a file that is `#included`.

We're also doing work that could potentially reducing the number of files processed (for some future release)
